app/Console/Commands/OneOff/FixDuplicateDemographics.php

```diff
@@ -29,32 +29,34 @@ public function handle()
     {
         $workdays = [];
 
-        WorkdayDemographic::selectRaw('workday_id, type, subtype, name, count(*) as num, sum(amount) as sum')
+        Demographic::selectRaw('demographical_id, type, subtype, name, count(*) as num, sum(amount) as sum')
             // group by is case insensitive, so in order to avoid turning up false positives, we cast to binary
             // to get the DB to recognize different casing.
-            ->groupByRaw('workday_id, type, subtype, name, Cast(name as binary)')
+            ->groupByRaw('demographical_id, type, subtype, name, Cast(name as binary)')
+            ->where('demographical_type', Workday::class)
             ->orderByRaw('num desc')
             ->chunk(10, function ($chunk) use (&$workdays) {
                 foreach ($chunk as $demographic) {
                     if ($demographic->num == 1) {
                         return false;
                     }
 
-                    $demographic['rows'] = WorkdayDemographic::where([
-                        'workday_id' => $demographic->workday_id,
+                    $demographic['rows'] = Demographic::where([
+                        'demographical_id' => $demographic->demographical_id,
+                        'demographical_type' => Workday::class,
                         'type' => $demographic->type,
                         'subtype' => $demographic->subtype,
                         'name' => $demographic->name,
                     ])->select('id', 'amount')->get()->toArray();
-                    $workdays[$demographic->workday_id][] = $demographic->toArray();
+                    $workdays[$demographic->demographical_id][] = $demographic->toArray();
                 }
 
                 return true;
             });
 
         foreach ($workdays as $workdayId => $stats) {
             foreach ($stats as $stat) {
-                $workday = Workday::find($workdayId);
+                $workday = Workday::withTrashed()->find($workdayId);
```

Comment:
Why is withTrashed neccessary here?

Reply:
Because the query that pulls demographics above can pull demographics in trashed workdays, and since the code is already here to fix them, it seems worthwhile to fix them in case the workdays are ever resurrected.